git-annex-remote-rclone

```diff
@@ -198,24 +198,20 @@ while read -r line; do
 			# Any nonzero object count means present.
 			# Some rclone backends support multiple
 			# files under one name.
-			if check_result=$(rclone size "$LOC$key" 2>&1) &&
-			    echo $check_result|grep -E 'Total objects: [1-9][0-9]* Total size' >&2 &&
-			    ! echo $check_result|grep 'Total size: 0 (0 bytes)' >&2; then
-				echo CHECKPRESENT-SUCCESS "$key"
-			else
-				# rclone 1.29 used 'Total objects: 0'
-				# rclone 1.30 uses 'directory not found'
-				if echo $check_result|grep 'Total objects: 0' >&2 || 
-				   echo $check_result|grep ' directory not found' >&2; then 
-					echo CHECKPRESENT-FAILURE "$key"
+			if check_result=$(rclone size --json "$LOC$key" 2>&1) &&
+			    echo "$check_result" | grep '"count":' >/dev/null; then
+				if echo "$check_result" | grep -E '"count":\s*[1-9][0-9]*' >/dev/null; then
+					echo CHECKPRESENT-SUCCESS "$key"
 				else
-					# When the directory does not exist,
-					# the remote is not available.
-					# (A network remote would similarly
-					# fail with CHECKPRESENT-UNKNOWN
-					# if it couldn't be contacted).
-					echo CHECKPRESENT-UNKNOWN "$key" "remote currently unavailable or git-annex-remote-rclone failed to parse rclone output"
+					echo CHECKPRESENT-FAILURE "$key"
 				fi
+			else
```

Comment:
When the directory is non existent `check_result` value is `Failed to size: directory not found`. This will make the remote return `CHECKPRESENT-UNKNOWN` instead of `CHECKPRESENT-FAILURE`

Reply:
Also I think this is an upstream issue because we asked for json output and instead we get an empty output and a human-readable stderr, which is kind the opposite of the idea of a `--json` flag usage

Reply:
Hmm, I can’t reproduce this or I don’t understand the situation you are describing. What do you mean by “the directory is non existent”? Here is what I am getting:

```
› rclone size --json b2:kirelagin-annex/foo/bar/baz
{"count":0,"bytes":0}
```

Reply:
@kirelagin can you try with a backend that is not an object storage? This is rclone being inconsistent in the output

```
$ rclone size --json zarel-blaze:test-bucket-annex/yay/yayy
{"count":0,"bytes":0}

$ rclone size --json zarel-box:thisfolderdoesnotexist
2022/04/20 20:51:16 ERROR : : error listing: directory not found
2022/04/20 20:51:16 Failed to size with 2 errors: last error was: directory not found

$ rclone size --json zarel-drive:test/folder/file
2022/04/20 20:51:35 Failed to size: directory not found

```

Reply:
Yes, you are right, I am getting this with a non-object backend too. I am not sure what the guarantees of the JSON output are, but, I guess, we can assume that if the exit code is >0, then we are not getting JSON.